### PR TITLE
Added hyperlinks to state and props + Arrow function definition

### PR DESCRIPTION
**Pull request by Emiliano-Russo**
### Description

components file was edited

state and props now have hyperlinks to their own page
props argument added to function component
arrow function + explanation added

Fixes issue #414

### Type of Change

- Adding a new entry: yes added arrow function
- Updating the documentation: yes, no props argument on function

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

### CODE REVIEW

**Comment by Dusch4593** [comment]
Hi, @Emiliano-Russo! Thanks for helping us out with this entry; we truly appreciate it! I'll take a look at the changes you recently pushed and leave some comments and/or suggestions for you to resolve afterwards. Look out for those within the next 24 hours 😄

File: content/react/concepts/components/components.md [hunk 1]
```diff
@@ -13,8 +13,8 @@ CatalogContent:
 
 In React, components make up the UI of the application. Components are pieces of code that are reuseable and independent. Each component comes with 1 or 2 of the following things:
 
-- `state`: Data in a component expected to change over time
-- `props`: Data that can be shared from parent component to child component
+- [state](https://www.codecademy.com/resources/docs/react/state) : Data in a component expected to change over time 
+- [props](https://www.codecademy.com/resources/docs/react/props) : Data that can be shared from parent component to child component
```
Comment (Dusch4593) [inline content/react/concepts/components/components.md:17]:
```suggestion
- A [`state` object](https://www.codecademy.com/resources/docs/react/state) that contains component data that is expected to change over time. 
- A [`props` object](https://www.codecademy.com/resources/docs/react/props) with data that can be passed down from parent component to child component.
```

Let's keep the back-ticks around "state" and "props" and treat these list items as complete sentences.

File: content/react/concepts/components/components.md [hunk 2]
```diff
@@ -13,8 +13,8 @@ CatalogContent:
 
 In React, components make up the UI of the application. Components are pieces of code that are reuseable and independent. Each component comes with 1 or 2 of the following things:
```
Comment (Dusch4593) [inline content/react/concepts/components/components.md:14]:
```suggestion
In React, components make up the UI of the application. Components are pieces of code that are reusable and independent. Each component comes with one or both of the following:
```

File: content/react/concepts/components/components.md [hunk 3]
```diff
@@ -45,7 +45,20 @@ class ClassComponent extends React.Component {
...
-function FunctionComponent() {
+function FunctionComponent(props) {
+  return (
+    <div>
+      ...
+    <div>
+  )
+}
+```
+## Arrow function
```
Comment (Dusch4593) [inline content/react/concepts/components/components.md:56]:
```suggestion

## Arrow Function

```

Let's use "Title Case" for the headers and don't forget to surround them with an empty line. 😃

File: content/react/concepts/components/components.md [hunk 4]
```diff
@@ -45,7 +45,20 @@ class ClassComponent extends React.Component {
...
+
+ ES6 version of JavaScript
```
Comment (Dusch4593) [inline content/react/concepts/components/components.md:58]: Can we turn into a complete sentence ending with a period? Maybe we could also include a hyperlink to the Docs entry for arrow functions? 

https://www.codecademy.com/resources/docs/javascript/arrow-functions

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@Emiliano-Russo I've finished my review! 😃 Feel free to take a look at my comments/suggestions and tag me once you're finished. If you have any questions along the way, we're here to help! 👍

**Review by Dusch4593 — APPROVED** [review]
Hi, @Emiliano-Russo thanks for the changes! I'm going to approve this and make a quick change to the latter part of the file. Should we conduct one more review, @sonnynomnom?

**Comment by Dusch4593** [comment]
@Emiliano-Russo 

I made some edits in the following:

The meta `Description` tag:

- The `Description` meta tag near the top should (ideally) be 150 characters or less as this makes the web page for the entry more SEO-friendly.

The "Array Functions" section:

Technically, ESx isn't a "version of JavaScript". ECMAScript is the official name of the syntax specification most JS programmers use today. 

Also, let's avoid using 1st- and 2nd-person pronouns. Docs should have an inviting yet manual-like voice.

Thank you again for your work on this. I truly appreciate your help with this entry. 😃

**Review by lukeecart — APPROVED** [review]

**Comment by sonnynomnom** [comment]
merging!

**Comment by sonnynomnom** [comment]
Added two small changes: https://github.com/Codecademy/docs/commit/862dcb4678829eef332822fc06d274ad5d7cf92f

@Emiliano-Russo @Dusch4593 

Mostly just cus this looks funky:

Screen Shot 2021-11-30 at 4 46 39 PM

**Comment by sonnynomnom** [comment]
But anyway your entry is live!

**Pull request merged**